### PR TITLE
chore(search-indexes): set up search indexes detection COMPASS-7196

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by kmruiz — APPROVED** [review]

File: packages/instance-model/lib/model.js [hunk 1]
```diff
@@ -256,6 +257,33 @@ const InstanceModel = AmpersandModel.extend(
...
+    async getIsSearchSupported({
+      ns,
+      dataService,
+      force = false
+    }) {
+      if (this.isSearchIndexesSupported === undefined || force) {
+        try {
+          const isSearchIndexesSupported = await dataService.isListSearchIndexesSupported(ns);
+          this.set({ isSearchIndexesSupported });
+        } catch (e) {
```
Comment (lerouxb) [inline packages/instance-model/lib/model.js:279]: Why this try/catch? isListSearchIndexesSupported's entire body is already a try/catch?

File: packages/collection-model/lib/model.js [hunk 2]
```diff
@@ -268,12 +269,31 @@ const CollectionModel = AmpersandModel.extend(debounceActions(['fetch']), {
...
+     * With this setup, when a user opens the first collection, we set this property
+     * on the instance model and then from there its value is read avoiding call to server.
+     */
+    try {
+      isSearchIndexesSupported = await getParentByType(this, 'Instance')
+        .getIsSearchSupported({
+          dataService,
+          ns: this.ns,
+        });
+    } catch (e) {
```
Comment (lerouxb) [inline packages/collection-model/lib/model.js:286]: Why this try/catch? How would this fail?

Comment (mabaasit) [reply packages/collection-model/lib/model.js:286]: both of these would not fail as such. but if it even decides to fail, i don't want this check to break the app as the methods calling this function do not have error handling.

Comment (gribnoysup) [reply packages/collection-model/lib/model.js:286]: We probably shouldn't ignore errors like that, if this starts unexpectedly throwing for any reason, we want this to be a signal that we see through a failing test (or worst case scenario user report)

Comment (mabaasit) [reply packages/collection-model/lib/model.js]: removed in [3ed908d](https://github.com/mongodb-js/compass/pull/4815/commits/3ed908db719f5342d3f1d7354fbdd93b5aa11119)

File: packages/compass-indexes/src/modules/search-indexes.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    };
+  }
+
+  return state;
+}
+
+export const setStatus = (status: SearchIndexesStatus): SetStatusAction => ({
+  type: ActionTypes.SetStatus,
+  status,
+});
```
Comment (gribnoysup) [inline packages/compass-indexes/src/modules/search-indexes.ts:56]: Setter actions are an an anti-pattern, currently this is only initial value where this is being used, can we pass it to the store as initial state?

Comment (mabaasit) [reply packages/compass-indexes/src/modules/search-indexes.ts:56]: Ended up with https://github.com/mongodb-js/compass/pull/4821 PR to handle it separately.

**Pull request merged**